src/webapi/events/pointer.rs

```diff
@@ -0,0 +1,483 @@
+use webcore::value::Reference;
+use webcore::try_from::TryInto;
+use webapi::event::{IEvent, IUiEvent, UiEvent, Event, ConcreteEvent};
+use webapi::events::mouse::{IMouseEvent, MouseEvent, MouseButton};
+
+/// The `IPointerEvent` interface represents the state of a DOM event produced by a pointer
+/// such as the geometry of the contact point, the device type that generated the event, the
+/// amount of pressure that was applied on the contact surface, etc.
+///
+/// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent)
+// https://w3c.github.io/pointerevents/#pointerevent-interface
+// https://w3c.github.io/pointerevents/extension.html#extensions-to-the-pointerevent-interface
+pub trait IPointerEvent: IMouseEvent {
+    /// Returns a unique identifier for the pointer causing the event.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent/pointerId)
+    // https://w3c.github.io/pointerevents/#dom-pointerevent-pointerid
+    #[inline]
+    fn pointer_id( &self ) -> i32 {
+        js!(
+            return @{self.as_ref()}.pointerId;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the width, in CSS pixels, of the contact geometry of the pointer.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent/width)
+    // https://w3c.github.io/pointerevents/#dom-pointerevent-width
+    #[inline]
+    fn width( &self ) -> f64 {
+        js!(
+            return @{self.as_ref()}.width;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the height, in CSS pixels, of the contact geometry of the pointer.
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent/height)
+    // https://w3c.github.io/pointerevents/#dom-pointerevent-height
+    #[inline]
+    fn height( &self ) -> f64 {
+        js!(
+            return @{self.as_ref()}.height;
+        ).try_into().unwrap()
+    }
+
+    /// Returns the normalized pressure of the pointer in the range [0, 1]
+    ///
+    /// [(JavaScript docs)](https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent/pressure)
+    // https://w3c.github.io/pointerevents/#dom-pointerevent-pressure
+    #[inline]
+    fn pressure( &self ) -> f64 {
```

Comment:
Hmm... looking at the specs this should return `f32`, I guess, although right now `try_into` doesn't support it. Can you leave a `TODO` so that we can change it later?